src/Futhark/Internalise.hs

```diff
@@ -1764,13 +1766,14 @@ isOverloadedFunction qname args loc = do
         _ ->
           error "Futhark.Internalise.internaliseExp: non-int/bool type in Complement"
 
-    scatterF a si v desc = do
-      si' <- letExp "write_si" . BasicOp . SubExp =<< internaliseExp1 "write_arg_i" si
+    scatterF dim a si v desc = do
+      si' <- do
+        sis <- internaliseExp "write_arg_i" si
+        mapM (letExp "write_si" . BasicOp . SubExp) sis
```

Comment:
Why not `internaliseExpToVars` here as well?

Reply:
No particular reason. Will change.